src/index.ts

```diff
@@ -271,7 +269,11 @@ export async function verify(token: string, secret: string, options: JwtVerifyOp
     }
     let keyFormat = 'raw'
     let keyData
-    if (secret.startsWith('-----BEGIN')) {
+    if (typeof secret === 'object') {
```

Comment:
According to the type definitions `secret` can only be a `string`. This should be typed correctly.